src/formatDate.js

```diff
@@ -50,7 +50,31 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [, , , fromSeparator] = fromFormat;
+  const [, , , toSeparator] = toFormat;
+  const dateArr = date.split(fromSeparator);
+  const dateKeys = {};
+  const finalView = [];
+
+  for (let i = 0; i < dateArr.length; i++) {
+    if (fromFormat[i] === 'YY') {
+      const front = dateArr[i] < 30 ? 20 : 19;
+
+      dateKeys['YYYY'] = `${front}${dateArr[i]}`;
+    }
+
+    if (fromFormat[i] === 'YYYY') {
+      dateKeys['YY'] = dateArr[i].slice(2);
+    }
```

Comment:
now you are doing  dateKeys[fromFormat[i]] = dateArr[i]; this part in every cases and it can rewrite your previous actions
```suggestion
    if (fromFormat[i] === 'YY') {
      const front = dateArr[i] < 30 ? 20 : 19;

      dateKeys['YYYY'] = `${front}${dateArr[i]}`;
      continue;
    }

    if (fromFormat[i] === 'YYYY') {
      dateKeys['YY'] = dateArr[i].slice(2);
      continue;
    }
```

Reply:
@maxim2310, but when I put "continue" for the 2nd "if" 4 tests are failing, and as a result, there is no year received. I assume you meant to add "continue" only for the 1st "if". Correct me if I'm wrong.

Reply:
@julmak I suggest you to use switch case statement in this case